### PR TITLE
added an option to enable GCP healthcheck endpoints

### PR DESCRIPTION
**Pull request by timothy-spencer**
## Description

This adds an option that lets you enable GCP healthcheck endpoints.
https://cloud.google.com/appengine/docs/flexible/custom-runtimes/configuring-your-app-with-app-yaml#updated_health_checks

It also incorporates suggestions from https://github.com/pusher/oauth2_proxy/pull/93 that lets it work with GKE ingresses.

## Motivation and Context

GCP likes to know when the app is up and running.  Enabling these endpoints makes it so that it is easy to run the proxy in the GCP App Engine environment

## How Has This Been Tested?

I use the go test ./... tests, as well as tested this by hand by launching with and without the flag enabled and attempting to go to locahost:4180/liveness_check and localhost:4180/readiness_check.  With the flag enabled, I got back "OK", and without, I got the normal auth redirect.

## Checklist:

- [X] My change requires a change to the documentation or CHANGELOG.
- [x] I have updated the documentation/CHANGELOG accordingly.
- [X] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

**Comment by ploxiln** [comment]
Have you considered using `-skip-auth-regex` to allow these two paths, and implementing the health check paths in the reverse-proxied application?

**Comment by timothy-spencer** [comment]
@ploxiln Hmm.  Interesting idea.  I think it's not quite the right approach, though, because if the upstream app were to get sad, GCP would begin killing and restarting the proxy, even if it were actually operating perfectly.  I'd rather the proxy remain up and happy, waiting for the upstream to return.

**Comment by benfdking** [comment]
Looks good to me

File: main.go [hunk 1]
```diff
@@ -139,8 +141,14 @@ func main() {
 
 	rand.Seed(time.Now().UnixNano())
 
+	var myhandler http.Handler
```
[the remark this answers is not in the document]
Comment (timothy-spencer) [reply main.go]: Yeah, I always feel a bit weird being so bold as to use something that feels like a keyword or something that would be used elsewhere, thus my attempt to make variables like this "mine".  :-)  But I know that's my own silliness.  Will do.

File: README.md [hunk 2]
```diff
@@ -48,7 +48,7 @@ Valid providers are :
 - [GitHub](#github-auth-provider)
 - [GitLab](#gitlab-auth-provider)
 - [LinkedIn](#linkedin-auth-provider)
-- [login.gov](#login.gov-provider)
+- [login.gov](#logingov-provider)
```
[the remark this answers is not in the document]
Comment (timothy-spencer) [reply README.md]: It just bugged me.  Glad you aren't unhappy about it!

**Pull request merged**